signac/core/syncedattrdict.py

```diff
@@ -17,10 +17,48 @@
 from ..errors import KeyTypeError
 
 
+class AttrDictFormatValidator:
+    """Implement the validation for :class:`SyncedAttrDict`."""
+
+    VALID_KEY_TYPES = (str, int, bool, type(None))
+
+    @classmethod
+    def validate(cls, data):
+        """Raise an exception if data is invalid.
+
+        Parameters
+        ----------
+        data:
+            Data to validate.
+
+        Returns
+        -------
+        data
+
+        Raises
+        ------
+        KeyTypeError:
+            If keys have unsupported data type.
+        InvalidKeyError
+            If key is not supported.
+        """
+        if isinstance(data, Mapping):
+            for key in data.keys():
+                if isinstance(key, str):
+                    if '.' in key:
+                        raise InvalidKeyError(
+                            "SyncedAttrDict keys may not contain dots ('.'): {}".format(key))
+                elif not isinstance(key, cls.VALID_KEY_TYPES):
+                    raise KeyTypeError(
+                        "SyncedAttrDict keys must be str, int, bool or None, not {}"
+                        .format(type(key).__name__))
+        return data
+
+
 class SyncedAttrDict(SyncedCollection, MutableMapping):
     """Implement the dict data structure along with values access through attributes named as keys.
 
-    The SyncedAttrDict inherits from :class:`~core.collection_api.SyncedCollection`
+    The SyncedAttrDict inherits from :class:`~core.synced_colletion.SyncedCollection`
```

Comment:
```suggestion
    The SyncedAttrDict inherits from :class:`~core.synced_collection.SyncedCollection`
```

Reply:
Sorry, I will change that.

Reply:
No problem, you can directly apply this suggestion.